src/main/scala/com/fulcrumgenomics/phasing/HapCutToVcf.scala

```diff
@@ -0,0 +1,413 @@
+
+  val PhaseSetFormatTag = "PS"
+  val PhaseSetFormatDescription = "Phase set for the genotype (position of the first variant)"
+  val PhaseSetFormatHeaderLine = new VCFFormatHeaderLine(PhaseSetFormatTag, 1, VCFHeaderLineType.Integer, PhaseSetFormatDescription)
+
+  /** Parses the genotype information produced by HapCut. */
+  def apply(info: String): HapCutCallGenotypeInfo = {
+    val tokens = info.split(":")
+    new HapCutCallGenotypeInfo(
+      readCounts  = tokens(0).split(",").map(_.toInt).toList,
+      likelihoods = tokens(1).split(",").map(_.toFloat).toList,
+      delta       = tokens(2).toFloat,
+      rMEC        = tokens(3).toFloat
+    )
+  }
+
+  /** The VCF header FORMAT keys that will be added for HapCut specific-genotype information. */
+  val formatHeaderKeys: Seq[String] = {
+    Seq(ReadCountFormatTag, LikelihoodChangeFormatTag, MaxLikelihoodChangeFormatTag, MecReductionFormatTag, PhaseSetFormatTag)
+  }
+
+  /** The VCF header FORMAT lines that will be added for HapCut specific-genotype information. */
+  val formatHeaderLines: Seq[VCFHeaderLine] = {
+    Seq(ReadCountFormatHeaderLine, LikelihoodChangeFormatHeaderLine, MaxLikelihoodChangeFormatHeaderLine, MecReductionFormatHeaderLine, PhaseSetFormatHeaderLine)
+  }
+
+  /** Adds the HapCut-specific genotype information to the given genotype builder. */
+  def addTo(builder: GenotypeBuilder, info: HapCutCallGenotypeInfo): GenotypeBuilder = {
+    builder
+      .attribute(ReadCountFormatTag, info.readCounts.asJava)
+      .attribute(LikelihoodChangeFormatTag, info.likelihoods.asJava)
+      .attribute(MaxLikelihoodChangeFormatTag, info.delta)
+      .attribute(MecReductionFormatTag, info.rMEC)
+  }
+}
+
+/** Information about a variant phased by HapCut.
+  *
+  * @param offset the 1-based offset of the variant relative to the variants given to HapCut.
+  * @param haplotype1 the first allele.
+  * @param haplotype2 the second allele.
+  * @param contig the contig.
+  * @param pos the position.
+  * @param ref the reference allele.
+  * @param alts the alternate alleles.
+  * @param genotype the genotype.
+  * @param info the HapCut-specific genotype information.
+  * @param firstInBlock true if this variant is the first variant in a phased block, false otherwise.
+  */
+private class HapCutCall private(val offset: Int,
+                                 val haplotype1: Int, val haplotype2: Int,
+                                 val contig: String, val pos: Int,
+                                 val ref: String, val alts: String,
+                                 val genotype: String, val info: HapCutCallGenotypeInfo,
+                                 val firstInBlock: Boolean
+                                )
+
+private object HapCutCall {
+
+  /** Parse a variant line. */
+  def apply(callLine: String, blockOffset: Int): HapCutCall = {
+    val tokens = callLine.split("\t")
```

Comment:
This may end up being rather slow as it's a regex split on a single character.  You might want to try `htsjdk`'s `StringUtil.split()` instead.